xcm/xcm-builder/src/currency_adapter.rs

```diff
@@ -138,6 +141,7 @@ impl<
 		if let Some(amount) = Matcher::matches_fungible(what) {
 			if let Some(checked_account) = CheckedAccount::get() {
 				Currency::deposit_creating(&checked_account, amount);
+				Currency::deactivate(amount);
```

Comment:
both currency calls, withdraw and deposit_creating return imbalance, can be used to determine the reactivate/deactivate amount

Reply:
it should be equivalent.